File: client/src/app/cash-flow/models/cash-flow.ts
import { RecurringTransaction } from "src/app/transaction/models/transaction-recurring";

export interface CashFlow {
    transactions: RecurringTransaction[];
    totalEntradasPrevistas: number;
    totalEntradasRealizadas: number;
    totalSaidasPrevistas: number;
    totalSaidasRealizadas: number;
    totalSaldoPrevisto: number;
    totalSaldoRealizado: number;
}

File: server/src/main/java/br/edu/utfpr/model/CashFlow.java
package br.edu.utfpr.model;

import lombok.Data;

import java.math.BigDecimal;
import java.util.List;

@Data
public class CashFlow {

    private List<RecurringTransaction> transactions;
    private BigDecimal totalEntradasPrevistas;
    private BigDecimal totalEntradasRealizadas;
    private BigDecimal totalSaidasPrevistas;
    private BigDecimal totalSaidasRealizadas;
    private BigDecimal totalSaldoPrevisto;
    private BigDecimal totalSaldoRealizado;
}
